src/software/guides/addin-storage.md

```diff
@@ -0,0 +1,384 @@
+  }
+});
+```
+
+**Important Notes**
+
+Each invocation of the Add operation will create a new AddInData object with a unique Id bound to the entered AddInData GUID. The Id of the AddInData object is required to remove the object with the **Remove** method. See below for an example.
+
+The value of the Data property is a JSON object converted to a JSON String or a string containing the JSON format with escaped quotes (\"). This JSON string does not support single quotes or newlines. A convenient way to create the escaped string from a JSON object is to use the [JSON.stringify()](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) method.
+
+**Example 1** : This method call will correctly save the sample JSON and associate it to the Add-In with the GUID of a2C4ABQuLFkepPVf6-4OKAQ.
+
+```javascript
+api.call("Add",
+{
+  "typeName": "AddInData",
+  "entity": {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "groups": [{
+        "id": "GroupCompanyId"
+    }],
+    "details": JSON.stringify({
+        "date": "2016-01-01T00:00:00.000Z",
+        "items": [
+          {"name": "bottles", "price": 12},
+        ],
+        "customer": {"name": "joesmith", "email": "joe@smith.com"}
+    })
+  }
+});
+```
+
+## Retrieving Stored AddInData Content
+
+AddInData uses a search object to query specific data using an object&#39;s path in the JSON.
+
+The AddInDataSearch properties are as follows:
+
+| **Properties** |
+| --- |
+| [Id]({{site.baseurl}}/software/api/reference/#T:Geotab.Checkmate.ObjectModel.Id): The standard Id for any Entity. |
+| [AddInId]({{site.baseurl}}/software/api/reference/#T:Geotab.Checkmate.ObjectModel.Id): Must be provided when searching for anything other than Id. |
+| [Groups]({{site.baseurl}}/software/api/reference/#T:Geotab.Checkmate.ObjectModel.Group): Used to define the scope of a row of Add-In data. Works the same as any other ObjectModel Entity. |
+| SelectClause (String): Used to filter the resulting rows based on the JSON content of the row. Works with the object path notation described in usage. Independent of WhereClause. |
+| WhereClause (String): Used to filter the resulting rows based on the JSON content of the row. Works with the object path and operator notation described in usage. Independent of SelectClause. |
+
+As an example, you can use the [API Runner tool](https://my.geotab.com/sdk/api/apiRunner.html) to perform GET operations that return one or more AddInData objects:
+
+**Example**  **2** : Get the emails of all customers who have an item with a price less than 15
+
+```javascript
+api.call("Get",
+{
+  "typeName": "AddInData",
+  "search": {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "selectClause": "customer.email",
+    "whereClause": "items.[].price < 15"
+  }
+});
+```
+
+This method call will return an array with a single AddInData object:
+
+```json
+"result": [
+  {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
```

Comment:
We no longer require addInId when getting by ID.

Reply:
Assuming you meant line https://github.com/Geotab/sdk/pull/107/files/db97c69209f7d5d95ec833366a75f3c8106d261f#diff-68f2e0189da77ea55f69ea716d10d0ace841f5581c22e243dc57fbfd3d0e606cR103. If so, I modified this comment to correctly reflect that it's optional